app2/src/lib/transfer/normal/steps/FillingStep.svelte

```diff
@@ -32,6 +31,38 @@ const {
 let isErrorModalOpen = $state(false)
 let isReceiverOpen = $state(false)
 
+const eff = Effect.gen(function*() {
+  const oracle = yield* PriceOracle.PriceOracle
+  const { universal_chain_id: id } = yield* transferData.sourceChain.pipe(
+    Effect.mapError(() => new Error("no source chain selected")),
+  )
+  return yield* oracle.of(id)
+})
+
+const fakePrice = runPromiseExit$(eff)
+
+const fakePriceDisplay = $derived(pipe(
+  fakePrice.current,
+  Option.map(Exit.match({
+    onFailure: Cause.prettyErrors,
+    onSuccess: (x) => JSON.stringify(x, null, 2),
+  })),
+  Option.getOrElse(() => "Loading price..."),
+))
+
+const realPrice = runPromiseExit$(
+  Effect.provide(eff, PriceOracle.PriceOracle.Pyth),
```

Comment:
Providing a layer or service breaks reactivity such that the effect only runs once (until HMR triggers).

Would be amazing to solve this, but it shouldn't impact us in the short-term given we will only use the global app runtime in practice.